newrelic-agent/src/main/java/com/newrelic/agent/config/ObfuscateJvmPropsConfigImpl.java

```diff
@@ -0,0 +1,49 @@
+package com.newrelic.agent.config;
+
+import java.util.*;
+
+public class ObfuscateJvmPropsConfigImpl extends BaseConfig implements ObfuscateJvmPropsConfig {
+
+    private static final String SYSTEM_PROPERTY_ROOT="newrelic.config.obfuscate_jvm_props.";
+    private static final String ALLOW = "allow";
+    private static final String BLOCK = "block";
+    private static final String ENABLED = "enabled";
+    private static final boolean DEFAULT_ENABLED = true;
+    private static final Set<String> DEFAULT_ALLOW = new HashSet<>();
```

Comment:
Instead of using a HashSet, use a SingletonSet. For things that are going to ever be a single element, this allocates less memory and executes faster.
```suggestion
    private static final Set<String> DEFAULT_ALLOW = Collections.singleton("-X*");
```
If we ever add more to the list, then it will have to go back to your implementation.